### PR TITLE
SDL2_mixer SDL2 and related emus

### PR DESCRIPTION
**Pull request by Kelvfimer**
Updated SDL2_mixer on 64 and keeping lib32 to 2.0.4 (sorr seems not to be compatible with new version)
Updated SDL2 and added patch TMNT fix (Needed to check if the patch is needed for TMNT 1.0321 + DLC)
Updated cdogs and hurrican that had issues. (remember to delete cdgos old content, it's needed to download the game again, same with Hurrican)

Needed a new PR for the updated emus but let's wait until this PR is tested.

### CODE REVIEW

**Comment by shantigilbert** [comment]
This PR is stale. lots of stuff is not needed anymore:
e.g. the TMNTN SDL patch, all the "shared", libraries, etc. 

I would separate system changes vs emu updates.

File: packages/audio/fluidsynth/package.mk
Comment (shantigilbert) [inline packages/audio/fluidsynth/package.mk:1]: This change is not needed

File: packages/audio/libmodplug/package.mk
Comment (shantigilbert) [inline packages/audio/libmodplug/package.mk:1]: This change is not needed

File: packages/lib32/audio/lib32-fluidsynth/package.mk
Comment (shantigilbert) [inline packages/lib32/audio/lib32-fluidsynth/package.mk:1]: This package is not called anywhere. why is it needed?

File: packages/lib32/audio/lib32-libmodplug/package.mk
Comment (shantigilbert) [inline packages/lib32/audio/lib32-libmodplug/package.mk:1]: This change is not needed

**Comment by Kelvfimer** [comment]
Created 3 different prs

**Pull request closed without merging**